src/broadcast.jl

```diff
@@ -414,7 +421,11 @@ function Base.copyto!(
     A, C = C, A
   end
   if !isnothing(A) && !isnothing(C) && !isnothing(α) && !isnothing(β)
-    map!((r, t) -> β * r + α * t, T, T, A)
+
+    # map!((r, t) -> β * r + α * t, T, T, A)
+
+    ab = axpby{promote_type(typeof(α), typeof(β))}(α, β)
+    map!((r, t) -> ab(t, r), T, T, A)
```

Comment:
```suggestion
    map!(axpby(α, β), T, T, A)
```
with the struct rewritten as above I think you can just write it like this.

Reply:
In principle this should work. However, there are some issues with this as `axpby(α, β)` is not  inferred as a `Function` by the compiler. In this stack, this function should be called next
```
function map!(f::Function, R::ITensor, T1::ITensor, T2::ITensor)
  R !== T1 && error("`map!(f, R, T1, T2)` only supports `R === T1` right now")
  return settensor!(R, _map!!(f, tensor(R), tensor(T1), tensor(T2)))
end
```
but an error is thrown in this case.
I removed `::Function` from this `map!` and `_map!!` but this results in a scalar indexing issue on GPU.

Reply:
You can make the struct `axpy` a subtype of `Function`:
```julia
struct axpby{Alpha,Beta} <: Function
  alpha::Alpha
  beta::Beta
end
```
so you can keep the type restriction `f::Function` in those function calls but I'm not sure if that solves the scalar indexing issue you are seeing.

Reply:
Oh this worked perfectly and was much better than changing the lower map functions. The scalar indexing must have been related to `axpby` not being inferred as a `Function`. The code now runs properly, thanks!